### PR TITLE
[Feat](Nereids) Support use mv hint

### PR DESCRIPTION
**Pull request by LiBinfeng-01**
### What problem does this PR solve?

Issue Number: close #xxx

Related PR: #xxx

Problem Summary:

support use_mv and no_use_mv hint to control mv chose strategy in cbo
```explain select /*+ use_mv(mv1)*/ * from t1;```

### Release note

None

### Check List (For Author)

- Test 
    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [ ] No.
    - [x] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by hello-stephen** [comment]
Thank you for your contribution to Apache Doris.
Don't know what should be done next? See [How to process your PR](https://cwiki.apache.org/confluence/display/DORIS/How+to+process+your+PR).

Please clearly describe your PR:
1. What problem was fixed (it's best to include specific error reporting information). How it was fixed.
2. Which behaviors were modified. What was the previous behavior, what is it now, why was it modified, and what possible impacts might there be.
3. What features were added. Why was this function added?
4. Which code was refactored and why was this part of the code refactored?
5. Which functions were optimized and what is the difference before and after the optimization?

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

File: regression-test/suites/nereids_p0/hint/test_use_mv.groovy [hunk 1]
```diff
@@ -102,4 +128,147 @@ suite("test_use_mv") {
...
+    }
+    explain {
+        sql """select /*+ use_mv(mv1, mv3) */ c1 from t1"""
+        contains("Used: use_mv([mv1].[mv3] )")
+        contains("internal.test_cbo_use_mv.mv1 chose")
+        contains("internal.test_cbo_use_mv.mv3 not chose")
+    }
+    explain {
+        sql """select /*+ use_mv(mv3, mv1) */ c1 from t1"""
+        contains("Used: use_mv([mv3].[mv1] )")
```
Comment (seawinde) [inline regression-test/suites/nereids_p0/hint/test_use_mv.groovy:241]: should add `mv_rewrite_success` in Suite.groovy to check if use mv or not

File: regression-test/suites/nereids_p0/hint/test_use_mv.groovy [hunk 2]
```diff
@@ -102,4 +128,147 @@ suite("test_use_mv") {
...
+    }
+    explain {
+        sql """select /*+ use_mv(mv3) */ c1 from t1"""
+        contains("Used: use_mv([mv3])")
+        contains("internal.test_cbo_use_mv.mv3 chose")
+    }
+    explain {
+        sql """select /*+ no_use_mv(mv1, mv3) */ c1 from t1"""
+        contains("Used: no_use_mv([mv1].[mv3] )")
+        notContains("internal.test_cbo_use_mv.mv3")
```
Comment (seawinde) [inline regression-test/suites/nereids_p0/hint/test_use_mv.groovy:258]: notContains should use `mv_not_part_in` in Suite.groovy

File: regression-test/suites/nereids_p0/hint/test_use_mv.groovy [hunk 3]
```diff
@@ -102,4 +128,147 @@ suite("test_use_mv") {
...
+    def query4 = """select c3 from t3"""
+
+    async_create_mv(db, query1, "mv1")
+    async_create_mv(db, query2, "mv2")
+    async_create_mv(db, query3, "mv3")
+    async_create_mv(db, query4, "mv4")
+
+    sql """use ${db};"""
+    explain {
+        sql """select /*+ use_mv(mv1)*/ * from t1;"""
```
Comment (seawinde) [inline regression-test/suites/nereids_p0/hint/test_use_mv.groovy:161]: Maybe we should add three part names for async materialized view and for part names for sync materialized view test

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/InitMaterializationContextHook.java [hunk 4]
```diff
@@ -104,6 +107,56 @@ protected void doInitMaterializationContext(CascadesContext cascadesContext) {
...
+        }
+        return hintMTMVs;
+    }
+
+    private List<MaterializationContext> getMvIdWithNoUseMvHint(List<MaterializationContext> mtmvCtxs,
+                                                                    UseMvHint useMvHint) {
+        List<MaterializationContext> hintMTMVs = new ArrayList<>();
+        if (useMvHint.isAllMv()) {
+            useMvHint.setStatus(Hint.HintStatus.SUCCESS);
+            return hintMTMVs;
```
Comment (seawinde) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/InitMaterializationContextHook.java:127]: this means no mv can be written?

Comment (LiBinfeng-01) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/InitMaterializationContextHook.java:127]: yes

File: regression-test/suites/nereids_p0/hint/test_use_mv.groovy (hunk 1 above)
Comment (LiBinfeng-01) [reply regression-test/suites/nereids_p0/hint/test_use_mv.groovy:241]: replace by explain memo plan

File: regression-test/suites/nereids_p0/hint/test_use_mv.groovy (hunk 2 above)
Comment (LiBinfeng-01) [reply regression-test/suites/nereids_p0/hint/test_use_mv.groovy:258]: replace by explain memo plan

File: regression-test/suites/nereids_p0/hint/test_use_mv.groovy (hunk 3 above)
Comment (LiBinfeng-01) [reply regression-test/suites/nereids_p0/hint/test_use_mv.groovy]: done

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run p0

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Review by seawinde — APPROVED** [review]

**Review by englefly — APPROVED** [review]

**Pull request merged**